### ISSUE TITLE
Training Got Stuck

### ISSUE THREAD

**Issue by Wujingqian** [xinntao/BasicSR#269]
Hi! Thanks for your wonderful Tool Box.
I'm training ESRGAN with my own dataset. The problem right now is when the training gets to some point, it just got stuck and stop there forever.
![image](https://user-images.githubusercontent.com/30253939/90356351-4ab0de00-e01d-11ea-896e-17f4fc958112.png)
I tried to stop and re-run the program but it got stuck again.
Do you know what the problem is?

# general settings
name: My_ESRGAN
model_type: ESRGANModel
scale: 4
num_gpu: 1

# dataset and data loader settings
datasets:
  train:
    name: My_Train_Data
    type: PairedImageDataset
    dataroot_gt: /home/wjq/BasicSR/datasets/Train_Data
    dataroot_lq: /home/wjq/BasicSR/datasets/Train_Data_LR
    # (for lmdb)
    # dataroot_gt: datasets/DIV2K/DIV2K_train_HR_sub.lmdb
    # dataroot_lq: datasets/DIV2K/DIV2K_train_LR_bicubic_X4_sub.lmdb
    filename_tmpl: '{}'
    io_backend:
      type: disk
      # (for lmdb)
      # type: lmdb

    gt_size: 128
    use_flip: true
    use_rot: true

    # data loader
    use_shuffle: true
    num_worker: 6  # per GPU
    batch_size: 16  # total
    dataset_enlarge_ratio: 1000

  val:
    name: Set1
    type: PairedImageDataset
    dataroot_gt: /home/wjq/BasicSR/datasets/Test_Data
    dataroot_lq: /home/wjq/BasicSR/datasets/Test_Data_LR
    io_backend:
      type: disk

# network structures
network_g:
  type: RRDBNet
  num_in_ch: 3
  num_out_ch: 3
  num_feat: 64
  num_block: 23

network_d:
  type: VGGStyleDiscriminator128
  num_in_ch: 3
  num_feat: 64

# path
path:
  pretrain_model_g: /home/wjq/BasicSR/pre_trained/ESRGAN_PSNR_SRx4_DF2K_official-150ff491.pth
  strict_load: true
  resume_state: ~

# training settings
train:
  optim_g:
    type: Adam
    lr: !!float 1e-4
    weight_decay: 0
    betas: [0.9, 0.99]
  optim_d:
    type: Adam
    lr: !!float 1e-4
    weight_decay: 0
    betas: [0.9, 0.99]

  scheduler:
    type: MultiStepLR
    milestones: [50000, 100000, 200000, 300000]
    gamma: 0.5

  niter: 400000
  warmup_iter: -1  # no warm up

  # losses
  pixel_opt:
    type: L1Loss
    loss_weight: !!float 1e-2
    reduction: mean
  perceptual_opt:
    type: PerceptualLoss
    layer_weights:
      'conv5_4': 1  # before relu
    vgg_type: vgg19
    use_input_norm: true
    perceptual_weight: 1.0
    style_weight: 0
    norm_img: false
    criterion: l1
  gan_opt:
    type: GANLoss
    gan_type: vanilla
    real_label_val: 1.0
    fake_label_val: 0.0
    loss_weight: !!float 5e-3

  net_d_iters: 1
  net_d_init_iters: 0

  manual_seed: 0

# validation settings
val:
  val_freq: !!float 5e3
  save_img: true

  metrics:
    psnr: # metric name, can be arbitrary
      type: calculate_psnr
      crop_border: 4
      test_y_channel: false

# logging settings
logger:
  print_freq: 100
  save_checkpoint_freq: !!float 5e3
  use_tb_logger: false
  wandb:
    project: basicsr
    resume_id: ~

# dist training settings
dist_params:
  backend: nccl
  port: 29500

In Command line:
2020-08-15 15:49:51,927 INFO: Validation Set1
         # psnr: 26.2428

2020-08-15 15:50:30,390 INFO: [My_ES..][epoch:207, iter:  20,100, lr:(1.000e-04,)] [eta: 1 day, 18:13:17, time (data): 0.365 (0.002)] l_g_pix: 3.1516e-04 l_g_percep: 5.2308e-01 l_g_gan: 4.2322e-02 l_d_real: 2.6469e-04 l_d_fake: 2.4991e-04 out_d_real: 9.1384e+01 out_d_fake: 8.2920e+01
2020-08-15 15:51:08,890 INFO: [My_ES..][epoch:208, iter:  20,200, lr:(1.000e-04,)] [eta: 1 day, 18:12:08, time (data): 0.375 (0.002)] l_g_pix: 3.8449e-04 l_g_percep: 6.5796e-01 l_g_gan: 1.6233e-02 l_d_real: 5.6346e-02 l_d_fake: 5.7430e-02 out_d_real: 1.8667e+01 out_d_fake: 1.5477e+01
2020-08-15 15:51:47,540 INFO: [My_ES..][epoch:209, iter:  20,300, lr:(1.000e-04,)] [eta: 1 day, 18:11:03, time (data): 0.390 (0.002)] l_g_pix: 3.6667e-04 l_g_percep: 5.6971e-01 l_g_gan: 3.3218e-02 l_d_real: 4.9696e-03 l_d_fake: 2.8609e-03 out_d_real: 4.7469e+01 out_d_fake: 4.0829e+01
2020-08-15 15:52:26,426 INFO: [My_ES..][epoch:210, iter:  20,400, lr:(1.000e-04,)] [eta: 1 day, 18:10:02, time (data): 0.377 (0.002)] l_g_pix: 3.3959e-04 l_g_percep: 6.2266e-01 l_g_gan: 9.3960e-03 l_d_real: 2.0036e-01 l_d_fake: 1.9684e-01 out_d_real: 1.1326e+02 out_d_fake: 1.1158e+02
2020-08-15 15:53:05,349 INFO: [My_ES..][epoch:211, iter:  20,500, lr:(1.000e-04,)] [eta: 1 day, 18:09:02, time (data): 0.392 (0.003)] l_g_pix: 3.6478e-04 l_g_percep: 5.6358e-01 l_g_gan: 5.9727e-05 l_d_real: 5.0785e+00 l_d_fake: 5.0776e+00 out_d_real: 1.1463e+02 out_d_fake: 1.1969e+02
2020-08-15 15:53:43,798 INFO: [My_ES..][epoch:212, iter:  20,600, lr:(1.000e-04,)] [eta: 1 day, 18:07:54, time (data): 0.369 (0.002)] l_g_pix: 3.2772e-04 l_g_percep: 6.8206e-01 l_g_gan: 2.8830e-02 l_d_real: 7.8938e-03 l_d_fake: 5.6942e-03 out_d_real: 7.9204e+01 out_d_fake: 7.3445e+01
2020-08-15 15:54:22,410 INFO: [My_ES..][epoch:213, iter:  20,700, lr:(1.000e-04,)] [eta: 1 day, 18:06:49, time (data): 0.365 (0.002)] l_g_pix: 2.7091e-04 l_g_percep: 5.7709e-01 l_g_gan: 4.6470e-02 l_d_real: 1.3022e-04 l_d_fake: 2.8757e-04 out_d_real: 1.7692e+02 out_d_fake: 1.6762e+02
2020-08-15 15:55:01,462 INFO: [My_ES..][epoch:214, iter:  20,800, lr:(1.000e-04,)] [eta: 1 day, 18:05:52, time (data): 0.381 (0.002)] l_g_pix: 2.6993e-04 l_g_percep: 6.6164e-01 l_g_gan: 2.5775e-02 l_d_real: 7.1960e-03 l_d_fake: 8.3010e-03 out_d_real: 1.8196e+02 out_d_fake: 1.7681e+02
2020-08-15 15:55:41,001 INFO: [My_ES..][epoch:215, iter:  20,900, lr:(1.000e-04,)] [eta: 1 day, 18:05:05, time (data): 0.381 (0.002)] l_g_pix: 4.0271e-04 l_g_percep: 6.7428e-01 l_g_gan: 5.0484e-03 l_d_real: 5.5853e-01 l_d_fake: 5.6600e-01 out_d_real: 1.8052e+02 out_d_fake: 1.8007e+02
2020-08-15 15:56:20,733 INFO: [My_ES..][epoch:216, iter:  21,000, lr:(1.000e-04,)] [eta: 1 day, 18:04:21, time (data): 0.407 (0.002)] l_g_pix: 4.1401e-04 l_g_percep: 6.3268e-01 l_g_gan: 1.2538e-02 l_d_real: 9.6001e-02 l_d_fake: 9.9716e-02 out_d_real: 1.4309e+02 out_d_fake: 1.4068e+02
2020-08-15 15:56:59,900 INFO: [My_ES..][epoch:217, iter:  21,100, lr:(1.000e-04,)] [eta: 1 day, 18:03:26, time (data): 0.384 (0.003)] l_g_pix: 2.9294e-04 l_g_percep: 6.0412e-01 l_g_gan: 1.2963e-02 l_d_real: 1.1324e-01 l_d_fake: 1.1465e-01 out_d_real: 1.1996e+02 out_d_fake: 1.1748e+02
2020-08-15 15:57:38,994 INFO: [My_ES..][epoch:218, iter:  21,200, lr:(1.000e-04,)] [eta: 1 day, 18:02:31, time (data): 0.371 (0.002)] l_g_pix: 3.0351e-04 l_g_percep: 5.8559e-01 l_g_gan: 1.7479e-02 l_d_real: 2.5179e-01 l_d_fake: 3.2110e-01 out_d_real: 1.5686e+02 out_d_fake: 1.5365e+02
2020-08-15 15:58:18,938 INFO: [My_ES..][epoch:219, iter:  21,300, lr:(1.000e-04,)] [eta: 1 day, 18:01:50, time (data): 0.371 (0.002)] l_g_pix: 3.1911e-04 l_g_percep: 5.6024e-01 l_g_gan: 5.9989e-02 l_d_real: 1.2509e-05 l_d_fake: 5.5234e-05 out_d_real: 6.1854e+01 out_d_fake: 4.9856e+01
2020-08-15 15:58:59,674 INFO: [My_ES..][epoch:220, iter:  21,400, lr:(1.000e-04,)] [eta: 1 day, 18:01:25, time (data): 0.392 (0.002)] l_g_pix: 4.4574e-04 l_g_percep: 6.6248e-01 l_g_gan: 2.0084e-02 l_d_real: 5.6749e-02 l_d_fake: 1.2008e-01 out_d_real: 1.7914e+01 out_d_fake: 1.3985e+01
2020-08-15 15:59:39,945 INFO: [My_ES..][epoch:221, iter:  21,500, lr:(1.000e-04,)] [eta: 1 day, 18:00:50, time (data): 0.383 (0.002)] l_g_pix: 3.7848e-04 l_g_percep: 5.9993e-01 l_g_gan: 2.0190e-02 l_d_real: 2.1405e-02 l_d_fake: 1.9871e-02 out_d_real: 2.5354e+01 out_d_fake: 2.1337e+01
2020-08-15 16:00:20,409 INFO: [My_ES..][epoch:222, iter:  21,600, lr:(1.000e-04,)] [eta: 1 day, 18:00:19, time (data): 0.387 (0.002)] l_g_pix: 3.6791e-04 l_g_percep: 6.5162e-01 l_g_gan: 1.4069e-02 l_d_real: 1.1553e-01 l_d_fake: 1.3985e-01 out_d_real: -1.7718e+01 out_d_fake: -2.0404e+01
2020-08-15 16:00:59,740 INFO: [My_ES..][epoch:223, iter:  21,700, lr:(1.000e-04,)] [eta: 1 day, 17:59:28, time (data): 0.379 (0.004)] l_g_pix: 3.2504e-04 l_g_percep: 6.2134e-01 l_g_gan: 5.8201e-02 l_d_real: 1.9706e-05 l_d_fake: 8.8094e-05 out_d_real: 9.2416e+01 out_d_fake: 8.0776e+01
2020-08-15 16:01:39,946 INFO: [My_ES..][epoch:224, iter:  21,800, lr:(1.000e-04,)] [eta: 1 day, 17:58:52, time (data): 0.372 (0.002)] l_g_pix: 3.9092e-04 l_g_percep: 6.6898e-01 l_g_gan: 1.3057e-02 l_d_real: 9.7243e-02 l_d_fake: 1.0279e-01 out_d_real: 9.7516e+01 out_d_fake: 9.5004e+01
2020-08-15 16:02:19,761 INFO: [My_ES..][epoch:225, iter:  21,900, lr:(1.000e-04,)] [eta: 1 day, 17:58:10, time (data): 0.386 (0.002)] l_g_pix: 4.2438e-04 l_g_percep: 6.6756e-01 l_g_gan: 3.1206e-03 l_d_real: 8.3372e-01 l_d_fake: 8.4435e-01 out_d_real: 8.8996e+01 out_d_fake: 8.9211e+01
2020-08-15 16:03:02,725 INFO: [My_ES..][epoch:226, iter:  22,000, lr:(1.000e-04,)] [eta: 1 day, 17:58:21, time (data): 0.368 (0.001)] l_g_pix: 3.7992e-04 l_g_percep: 6.8192e-01 l_g_gan: 6.1138e-03 l_d_real: 3.8089e-01 l_d_fake: 3.8140e-01 out_d_real: 1.2656e+02 out_d_fake: 1.2571e+02
2020-08-15 16:03:41,902 INFO: [My_ES..][epoch:227, iter:  22,100, lr:(1.000e-04,)] [eta: 1 day, 17:57:28, time (data): 0.386 (0.002)] l_g_pix: 4.0449e-04 l_g_percep: 6.2189e-01 l_g_gan: 2.2486e-02 l_d_real: 1.4423e-02 l_d_fake: 1.3582e-02 out_d_real: 8.5756e+01 out_d_fake: 8.1273e+01

Thank you!

**Comment by xinntao** [comment]
@Wujingqian From the configuration and log, I can not tell the reason.

When getting stuck, you can press Ctrl+c, and to see what error it raises~

**Comment by Wujingqian** [comment]
Thank you! I think it's an issue with my device. I solved it.

**Comment by xinntao** [comment]
@Wujingqian OK~

BTW, I have updated BasicSR to a newer version (https://github.com/xinntao/BasicSR/releases/tag/v1.0.0). From this version, it will become more stable and we will add more new features based on this version. You may have an update. Thanks :smile:
